### DISCUSSION TITLE
ModuleNotFoundError: No module named 'setuptools.command.build'

### DISCUSSION THREAD

**Discussion by DannyVarod** [pypa/setuptools#3370] [Q&A]
Hi, with latest version of setuptools, some older packages now won't install and output the error:
ModuleNotFoundError: No module named 'setuptools.command.build'
This seems to have started on version 62.4.0
Was there a breaking change in this version?

**Comment by abravalheri** [comment]
Hi @DannyVarod, in theory there is no breaking change in this version.
Can you post an isolated reproduction here?

Please note that using `python setup.py install` is deprecated and users are suggested to move to `pip`.

Als, could you also please confirm which operating system/distribution/version you are using?

**Reply by DannyVarod** [reply]
This occurred on Linux Ubuntu 20.04.4 while running both `pip install -r requirements.txt` and `python setup.py sdist` from Python 3.6.
In Python 3.7+ projects this did not reproduce.

Perhaps for v62.4.0 the Python requirement should be >= 3.7 so older versions of Python will be limited to version 62.3.4 and down which work.

This reproduced on for specific project's build (Ubuntu) + on a specific dev's machine (OS X) for an older Python 3.6 based project.
I'll post a minimal reproduction when I have one.

**Reply by abravalheri** [reply]
> Perhaps for v62.4.0 the Python requirement should be >= 3.7 so older versions of Python will be limited to version 62.3.4 and down which work.

Hi @DannyVarod, this is already true nowadays: https://github.com/pypa/setuptools/blob/94955ceb80945f5b9880545030db1de7a539c38f/setup.cfg#L27

Please try to avoid using `python setup.py install` directly. The recommendation is to replace this command with  `pip install .` whenever possible.

**Reply by DannyVarod** [reply]
I went over the process again, I see the build checked 2 different requirements files (deployed to two different places) for conflicts, however, they were accidentally done in the same env, despite being dependent on different Python versions, so this seems to have caused this.

**Comment by yaegel** [comment]
I experienced the same issue (related, in my case, to py7zr).  The solution in my build was to upgrade pip (pip3 install --upgrade pip).  Hope that helps!
